### PR TITLE
feat: change revoke and assign behavior to turn on/off the trigger

### LINKED ISSUE #76: Partition revoke is stopping all quartz jobs in the application
When we have Kafka rebalance we are shutting down all quartz jobs and it could impact other jobs in the application.

We should re-think the quartz job creation and use for example UnscheduleJob instead of.

[Quartz documentation](https://quartznet.sourceforge.io/apidoc/3.0/html/)

### LINKED ISSUE #91: Update kafka flow retry to .NET 6
As the support of .Net Core 3.1 is ending on [13 of December](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core)  we need to migrate to .NET 6 any project using that version of .Net Core

### PR DESCRIPTION
**Pull request by martinhonovais**
# Description

The revoke partition was stopping all quartz jobs because we were shutting down the quartz. This pull request changed that behavior to just schedule or unschedule the jobs.

Fixes #76 
Fixes #91 

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist

-   [X] My code follows the style guidelines of this project
-   [X] I have performed a self-review of my own code
-   [X] I have added tests to cover my changes
-   [ ] I have made corresponding changes to the documentation

### Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: src/KafkaFlow.Retry.UnitTests/KafkaFlow.Retry/Durable/Polling/QueueTrackerTests.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            // act
+            queueTracker.ScheduleJob();
+
+            await WaitForSeconds(6).ConfigureAwait(false);
+
+            queueTracker.UnscheduleJob();
+
+            await WaitForSeconds(15).ConfigureAwait(false);
+
+            var mockITriggerProvider1 = new Mock<ITriggerProvider>();
```
Comment (jdromeiro) [inline src/KafkaFlow.Retry.UnitTests/KafkaFlow.Retry/Durable/Polling/QueueTrackerTests.cs:85]: Rename to mockTriggerProvider2, and rename the mockTriggerProvider as mockTriggerProvider1.

Do the same for the queueTracker and queueTracker1.

File: src/KafkaFlow.Retry.UnitTests/KafkaFlow.Retry/Durable/Polling/QueueTrackerTests.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            jobExecutionContexts.Add(context);
+
+            return Task.CompletedTask;
+        }
+    }
+
+    public class QueueTrackerTests
+    {
+        [Fact]
+        public async Task QueueTracker_Create_Success()
```
Comment (jdromeiro) [inline src/KafkaFlow.Retry.UnitTests/KafkaFlow.Retry/Durable/Polling/QueueTrackerTests.cs:28]: I think we can explain better what we're trying to achieve with this test by setting the name as:
`QueueTracker_ScheduleAndUnscheduleJobs_Success`

File: src/KafkaFlow.Retry.UnitTests/KafkaFlow.Retry/Durable/Polling/QueueTrackerTests.cs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            await WaitForSeconds(6).ConfigureAwait(false);
+
+            queueTracker1.UnscheduleJob();
+
+            await WaitForSeconds(15).ConfigureAwait(false);
+
+            jobExecutionContexts.Where(x => x.PreviousFireTimeUtc is null).Count().Should().Be(2);
+            jobExecutionContexts.Where(x => x.PreviousFireTimeUtc is null && x.Trigger.Key.Name == "Trigger").Count().Should().Be(1);
+            jobExecutionContexts.Where(x => x.PreviousFireTimeUtc is null && x.Trigger.Key.Name == "Trigger1").Count().Should().Be(1);
+            (
```
Comment (jdromeiro) [inline src/KafkaFlow.Retry.UnitTests/KafkaFlow.Retry/Durable/Polling/QueueTrackerTests.cs:116]: Maybe make it a little easier to understand by re-writting as:
```
 var timeBetweenJobExecutionsWhileJobWasUnscheduled = jobExecutionContexts
                    .Where(x => x.Trigger.Key.Name == "Trigger1")
                    .OrderBy(x => x.FireTimeUtc)
                    .First()
                    .FireTimeUtc
                    -
                jobExecutionContexts
                    .Where(x => x.Trigger.Key.Name == "Trigger")
                    .OrderBy(x => x.FireTimeUtc)
                    .Last()
                    .FireTimeUtc;

            timeBetweenJobExecutionsWhileJobWasUnscheduled.Should().BeGreaterThan(TimeSpan.FromSeconds(15));
```

**Review by jdromeiro — CHANGES_REQUESTED** [review]

File: src/KafkaFlow.Retry.UnitTests/KafkaFlow.Retry/Durable/Polling/QueueTrackerCoordinatorTests.cs [hunk 4]
```diff
@@ -58,26 +53,38 @@ public void QueueTrackerCoordinator_Initialize_Success()
         public void QueueTrackerCoordinator_Shutdown_Success()
```
Comment (jdromeiro) [inline src/KafkaFlow.Retry.UnitTests/KafkaFlow.Retry/Durable/Polling/QueueTrackerCoordinatorTests.cs:53]: rename to Unschedule

File: src/KafkaFlow.Retry.UnitTests/KafkaFlow.Retry/Durable/Polling/QueueTrackerCoordinatorTests.cs [hunk 5]
```diff
@@ -34,19 +32,16 @@ public void QueueTrackerCoordinator_Initialize_Success()
             queueTrackerFactory
```
Comment (jdromeiro) [inline src/KafkaFlow.Retry.UnitTests/KafkaFlow.Retry/Durable/Polling/QueueTrackerCoordinatorTests.cs:32]: Rename the test name to Schedule

**Review by jdromeiro — CHANGES_REQUESTED** [review]

File: src/KafkaFlow.Retry/Durable/Polling/QueueTracker.cs [hunk 6]
```diff
@@ -1,124 +1,149 @@
...
-
-            lock (internalLock)
+            try
             {
-                this.scheduler = StdSchedulerFactory.GetDefaultScheduler(cancellationToken).GetAwaiter().GetResult();
-            }
-
-            this.scheduler
-                .Start(cancellationToken)
-                .ConfigureAwait(false)
```
[the remark this answers is not in the document]
Comment (martinhonovais) [reply src/KafkaFlow.Retry/Durable/Polling/QueueTracker.cs:66]: Done

File: src/KafkaFlow.Retry/Durable/Polling/QueueTracker.cs [hunk 7]
```diff
@@ -1,124 +1,149 @@
...
+                    this.logHandler.Info(
+                        "PollingJob Scheduler Started",
+                        new
+                        {
+                            PollingId = this.retryDurablePollingDefinition.Id,
+                            CronExpression = this.retryDurablePollingDefinition.CronExpression
+                        });
+                }
+
+                var triggerAlreadyDefined = this.scheduler
```
[the remark this answers is not in the document]
Comment (martinhonovais) [reply src/KafkaFlow.Retry/Durable/Polling/QueueTracker.cs:79]: This is unneeded code, so we have decided to remove it.

**Review by jdromeiro — DISMISSED** [review]

File: src/KafkaFlow.Retry/Durable/Polling/JobDetailProvider.cs [hunk 8]
```diff
@@ -0,0 +1,63 @@
+﻿using Dawn;
+using KafkaFlow.Retry.Durable.Definitions;
+using KafkaFlow.Retry.Durable.Encoders;
+using KafkaFlow.Retry.Durable.Repository;
+using KafkaFlow.Retry.Durable.Repository.Adapters;
+using Quartz;
```
Comment (carlosgoias) [inline src/KafkaFlow.Retry/Durable/Polling/JobDetailProvider.cs:6]: using inside of namespance

**Review by carlosgoias — DISMISSED** [review]

**Review by spookylsm — APPROVED** [review]

**Review by sergioamribeiro — APPROVED** [review]

**Review by carlosgoias — APPROVED** [review]

**Review by brunohfgomes — APPROVED** [review]

**Pull request merged**